### PR TITLE
Add-AzVhd Convenience Layer

### PR DESCRIPTION
**Pull request by grizzlytheodore**
## Description
adding Add-AzVhd Convenience Layer: direct upload to managed disk, using Assembly Isolation

There is a small breaking change component:
Before, Add-AzVHD would upload any sized VHD, including sizes that are not aligned to N * Megabyte. 
However the user would get an error when they try to attach the VHD file to a VM because it is not sized properly.
With the changes I made, if the VHD file size is not aligned to N * MegaByte, it will try to resize it using Hyper-V, and if Hyper-V is not found, the user will get an error and ask to resize VHD manually. 
This is the breaking change, but it is failing early with better directions for customers. 

## Checklist

- [x] I have read the [_Submitting Changes_](../blob/master/CONTRIBUTING.md#submitting-changes) section of [`CONTRIBUTING.md`](../blob/master/CONTRIBUTING.md)
- [x] The title of the PR is clear and informative
- [x] The appropriate `ChangeLog.md` file(s) has been updated:
    - For any service, the `ChangeLog.md` file can be found at `src/{{SERVICE}}/{{SERVICE}}/ChangeLog.md`
    - A snippet outlining the change(s) made in the PR should be written under the `## Upcoming Release` header -- no new version header should be added
- [ ] The PR does not introduce [breaking changes](../blob/master/documentation/breaking-changes/breaking-changes-definition.md)
- [x] If applicable, the changes made in the PR have proper test coverage
- [ ] For public API changes to cmdlets:
    - [ ] a cmdlet design review was approved for the changes in [this repository](https://github.com/Azure/azure-powershell-cmdlet-review-pr) (_Microsoft internal only_)
        - {Please put the link here}
    - [ ] the markdown help files have been regenerated using the commands listed [here](../blob/master/documentation/development-docs/help-generation.md#updating-all-markdown-files-in-a-module)

### CODE REVIEW

**Comment by grizzlytheodore** [comment]
@BethanyZhou @isra-fel   addressed your comment. please merge

**Comment by isra-fel** [comment]
Resolved conflicts in changelog

**Comment by grizzlytheodore** [comment]
do the CI failures need to be resolved? @isra-fel  
or is it CI issues?

**Comment by grizzlytheodore** [comment]
@isra-fel  could you approve/merge?

File: src/Compute/Compute/ChangeLog.md [hunk 1]
```diff
@@ -37,6 +41,12 @@
 * Added new cmdlets:
     - Add-AzVmssRunCommand
     - Remove-AzVmssRunCommand
+    - Set-AzVMRunCommand
+    - Get-AzVMRunCommand
+    - Remove-AzVMRunCommand
+    - Set-AzVmssVMRunCommand
+    - Remove-AzVmssVMRunCommand
+    - Get-AzVmssVMRunCommand
```
Comment (isra-fel) [inline src/Compute/Compute/ChangeLog.md:49]: These are already on main branch. It's unlikely to cause any problem but I'll double check after merge

File: src/Compute/Compute/StorageServices/AddAzureVhdCommand.cs [hunk 2]
```diff
@@ -25,51 +25,102 @@
...
+using Microsoft.Samples.HyperV.Storage;
+using Microsoft.Samples.HyperV.Common;
+using System.Threading;
 
 namespace Microsoft.Azure.Commands.Compute.StorageServices
 {
     /// <summary>
     /// Uploads a vhd as fixed disk format vhd to a blob in Microsoft Azure Storage
     /// </summary>
     [GenericBreakingChange("This cmdlet will now try to resize the VHD file to (N * Mib + 512 bytes) using Hyper-V before uploading.\n If Hyper-V is not found, user will have to either enabled Hyper-V or resize manually before uploading.")]
```
Comment (isra-fel) [inline src/Compute/Compute/StorageServices/AddAzureVhdCommand.cs:45]: Please remove this warning as the breaking change is about to happen.

**Review by isra-fel — CHANGES_REQUESTED** [review]

File: src/Compute/Compute/ChangeLog.md (hunk 1 above)
Comment (grizzlytheodore) [reply src/Compute/Compute/ChangeLog.md]: took this out to not be making changes to older version changelog

File: src/Compute/Compute/StorageServices/AddAzureVhdCommand.cs (hunk 2 above)
Comment (grizzlytheodore) [reply src/Compute/Compute/StorageServices/AddAzureVhdCommand.cs]: resolved

**Comment by grizzlytheodore** [comment]
@isra-fel  addressed yours comments. please approve/merge !

**Review by isra-fel — APPROVED** [review]

**Pull request merged**